src/main/java/uk/gov/cslearning/record/api/ApiExceptionHandler.java

```diff
@@ -8,33 +9,53 @@
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
-import uk.gov.cslearning.record.dto.ValidationErrors;
-import uk.gov.cslearning.record.dto.factory.ValidationErrorsFactory;
+import uk.gov.cslearning.record.dto.ErrorDto;
+import uk.gov.cslearning.record.dto.factory.ErrorDtoFactory;
 import uk.gov.cslearning.record.exception.BookingNotFoundException;
+import uk.gov.cslearning.record.exception.EventNotFoundException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
 
 @ControllerAdvice
 public class ApiExceptionHandler {
     private static final Logger LOGGER = LoggerFactory.getLogger(ApiExceptionHandler.class);
 
-    private final ValidationErrorsFactory validationErrorsFactory;
+    private final ErrorDtoFactory errorDtoFactory;
 
-    public ApiExceptionHandler(ValidationErrorsFactory validationErrorsFactory) {
-        this.validationErrorsFactory = validationErrorsFactory;
+    public ApiExceptionHandler(ErrorDtoFactory errorDtoFactory) {
+        this.errorDtoFactory = errorDtoFactory;
     }
 
     @ResponseStatus(HttpStatus.BAD_REQUEST)
     @ExceptionHandler(MethodArgumentNotValidException.class)
-    protected ResponseEntity<ValidationErrors> handleMethodArgumentNotValidException(MethodArgumentNotValidException e) {
+    protected ResponseEntity<ErrorDto> handleMethodArgumentNotValidException(MethodArgumentNotValidException e) {
         LOGGER.error("Bad Request: ", e);
 
-        return ResponseEntity.badRequest().body(validationErrorsFactory.create(e.getBindingResult().getFieldErrors()));
+        List<String> errors = e.getBindingResult().getFieldErrors().stream()
+                .map(error -> String.join(": ", error.getField(), error.getDefaultMessage()))
+                .collect(Collectors.toList());
+
+        return ResponseEntity.badRequest().body(errorDtoFactory.create(HttpStatus.BAD_REQUEST, errors));
     }
 
     @ResponseStatus(HttpStatus.NOT_FOUND)
-    @ExceptionHandler(BookingNotFoundException.class)
-    protected ResponseEntity<ValidationErrors> handleBookingNotFoundException(BookingNotFoundException e) {
+    @ExceptionHandler({BookingNotFoundException.class, EventNotFoundException.class})
+    protected ResponseEntity handleNotFoundException(RuntimeException e) {
         LOGGER.error("Not Found: ", e);
 
         return ResponseEntity.notFound().build();
     }
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    protected ResponseEntity<ErrorDto> handleConstraintViolationException(ConstraintViolationException e) {
+        LOGGER.error("Bad Request: ", e);
+
+        ErrorDto error = errorDtoFactory.create(HttpStatus.BAD_REQUEST,
+                Collections.singletonList(String.join(", ", e.toString(), e.getSQLException().getMessage())));
```

Comment:
Info - Would this expose low level SQL contrainsts to the HTTP message ie: duplicate key?

Reply:
Yes.

Reply:
We can not have that unless the application was in **debug or non-production** mode.